### PR TITLE
[PBIOS-43] Loader Kit

### PR DESCRIPTION
**Pull request by RachelRadford21**
**What does this PR do?**
[PBIOS-43] Loader Kit

**Screenshots:** Screenshots to visualize your addition/change
Simulator Screenshot - iPhone 15 Pro Max - 2024-04-02 at 15 26 55

**How to test?** Steps to confirm the desired behavior:
4. See addition/change

### Checklist
- [x] **LABELS** - Add a label: `breaking`, `bug`, `improvement`, `documentation`, or `enhancement`. See [Labels](https://github.com/powerhome/playbook-apple/labels) for descriptions.
- [x] **RELEASES** - Add the appropriate label: `Ready for Testing` / `Ready for Release`
- [x] **TESTING** - Have you tested your story?

### CODE REVIEW

File: Sources/Playbook/Components/Loader/PBLoader.swift [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+extension PBLoader {
+  var loaderLabelView: some View {
+    HStack(spacing: Spacing.small) {
+      loaderView
+      loaderLabel
+    }
+    .pbFont(spinnerFont, color: .text(.light))
+  }
+  var loaderView: some View {
+    GeometryReader { geo in
```
Comment (isismsilva) [inline Sources/Playbook/Components/Loader/PBLoader.swift:60]: Why are you using geometry reader here?

File: Sources/Playbook/Components/Loader/PBLoader.swift [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        }
+      }
+      .onAppear {
+        Timer.scheduledTimer(withTimeInterval: spinnerSpeed, repeats: true) { _ in
+          dotIndex = (dotIndex + 1) % dotsCount
+          contentSize = geo.size
+        }
+      }
+    }
+    .frame(maxWidth: contentSize.width)
```
Comment (isismsilva) [inline Sources/Playbook/Components/Loader/PBLoader.swift:77]: Why is the component frame being set in here?

**Review by isismsilva — CHANGES_REQUESTED** [review]

**Comment by isismsilva** [comment]
Hey love. There are somethings that are more difficult to achieve using GeometryReader and offset. 
1 - Setting a background won't work.
If you apply a color background it wont show anything because when you use offset with some position value, the offset wont be considered as part of the view. 

Screenshot 2024-04-03 at 3 16 21 PM

2 - Alignment will behave unexpectedly.
The view will be aligned with the wanted center unless you set the correct value for it and dislocate the entire animation.

Screenshot 2024-04-03 at 3 17 04 PM

3 - Unexpected computed size.
The view size will also have an unexpected value, In this case it's considering the dot size + the distance from its center. Instead of considering the size of the entire component. 

Screenshot 2024-04-03 at 3 31 50 PM

File: Sources/Playbook/Components/Loader/PBLoader.swift [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  let dotSize: CGFloat
+  let circleRadius: CGFloat
+  let spinnerFont: PBFont
+  let spinnerSpeed: TimeInterval
+  let alignment: HorizontalAlignment
+  public init(
+    rotationAngle: Double = 0,
+    dotIndex: Int = 0,
+    contentSize: CGSize = .zero,
+    dotsCount: Int = 7,
```
Comment (isismsilva) [inline Sources/Playbook/Components/Loader/PBLoader.swift:26]: I believe it should be 8 dots.

File: Sources/Playbook/Components/Loader/PBLoader.swift [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    self.spinnerSpeed = spinnerSpeed
+    self.alignment = alignment
+  }
+  var body: some View {
+      loaderView
+  }
+}
+
+extension PBLoader {
+  var loaderView: some View {
```
Comment (isismsilva) [inline Sources/Playbook/Components/Loader/PBLoader.swift:49]: The indentation looks odd here. Can you take a look?

**Review by isismsilva — CHANGES_REQUESTED** [review]

File: Sources/Playbook/Components/Loader/PBLoader.swift (hunk 3 above)
Comment (RachelRadford21) [reply Sources/Playbook/Components/Loader/PBLoader.swift:26]: hey! I just pushed this so I could change branches. Im not done. Should have said something, sorry!

**Review by isismsilva — APPROVED** [review]

**Pull request merged**